SQLite.CodeFirst/Internal/Builder/CreateTableStatementBuilder.cs

```diff
@@ -19,13 +21,19 @@ public CreateTableStatementBuilder(EntitySet entitySet, AssociationTypeContainer
 
         public CreateTableStatement BuildStatement()
         {
+            var nonAutoincrementKeys = entitySet.ElementType.KeyMembers
+                .Where(km => (km as EdmProperty)?.GetCustomAnnotation<AutoincrementAttribute>() == null);
```

Comment:
Hey,

Thanks to your reply. Im happy to contribute.
Yes ... I unfortunetly I belive your approach is not possible as put in the Summary pt 4. Is the only way to enforce this behavior (http://sqlite.org/autoinc.html):

```
Summary

1.The AUTOINCREMENT keyword imposes extra CPU, memory, disk space, and disk I/O overhead and should be avoided if not strictly needed. It is usually not needed.

2.In SQLite, a column with type INTEGER PRIMARY KEY is an alias for the ROWID (except in WITHOUT ROWID tables) which is always a 64-bit signed integer.

3.On an INSERT, if the ROWID or INTEGER PRIMARY KEY column is not explicitly given a value, then it will be filled automatically with an unused integer, usually one more than the largest ROWID currently in use. This is true regardless of whether or not the AUTOINCREMENT keyword is used.

4.If the AUTOINCREMENT keyword appears after INTEGER PRIMARY KEY, that changes the automatic ROWID assignment algorithm to prevent the reuse of ROWIDs over the lifetime of the database. In other words, the purpose of AUTOINCREMENT is to prevent the reuse of ROWIDs from previously deleted rows.

```
I had luck in creating a model having a single Autoincrement attributed model, never the less the model had only one Primary Key (int) field and without foreign keys. Perhaps the Autoincrement Attribute should imply no other primary keys are  present. I will certainly have a look at the teamPlayer.